File: goal-keeper-backend/src/main/java/com/goalkeeper/api/dto/SignUpRequestDto.java
package com.goalkeeper.api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.util.Set;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SignUpRequestDto {

    @NotBlank
    @Size(min = 3, max = 20)
    private String username;

    @NotBlank
    @Size(max = 50)
    @Email(message = "The given email is not valid")
    private String email;

    @NotBlank
    @Size(max = 50)
    private String firstName;

    @NotBlank
    @Size(max = 50)
    private String lastName;

    private Set<String> role;

    @NotBlank
    @Size(min = 6, max = 40)
    private String password;
}

File: goal-keeper/src/data/dto/SignupRequestDto.ts
import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject('SignupRequestDto')
export default class SignupRequestDto {

    @JsonProperty('username', String)
    public username: string | undefined = undefined;

    @JsonProperty('firstName', String)
    public firstName: string | undefined = undefined;

    @JsonProperty('lastName', String)
    public lastName: string | undefined = undefined;

    @JsonProperty('email', String)
    public email: string | undefined = undefined;

    @JsonProperty('password', String)
    public password: string | undefined = undefined;
}
